server/leafnode.go

```diff
@@ -934,7 +953,13 @@ func (s *Server) initLeafNodeSmap(c *client) {
 	for isubj := range acc.imports.services {
 		ims = append(ims, isubj)
 	}
-	acc.mu.RUnlock()
+	// Create a unique subject that will be used for loop detection.
+	lds := acc.lds
+	if lds == _EMPTY_ {
+		lds = "lds." + nuid.Next()
```

Comment:
Maybe make lds prefix a const somewhere.

Reply:
Made it a constant.